It fails on 2021.3 (doesn't support the UITK version of the editor nor projectwide actions afaik) with the following compilation error:

`Packages/com.unity.inputsystem/InputSystem/Editor/ControlPicker/InputControlPathEditor.cs(165,13): error CS0103: The name 'InputActionsEditorSettingsProvider' does not exist in the current context`

--- a/Packages/com.unity.inputsystem/InputSystem/Editor/ControlPicker/InputControlPathEditor.cs
+++ b/Packages/com.unity.inputsystem/InputSystem/Editor/ControlPicker/InputControlPathEditor.cs
@@ -162,7 +162,9 @@ namespace UnityEngine.InputSystem.Editor
 
         private void ShowDropdown(Rect rect, SerializedProperty serializedProperty, Action modifiedCallback)
         {
+            #if UNITY_INPUT_SYSTEM_PROJECT_WIDE_ACTIONS
             InputActionsEditorSettingsProvider.SetIMGUIDropdownVisible(true, false);
+            #endif
             if (m_PickerDropdown == null)
             {
                 m_PickerDropdown = new InputControlPickerDropdown(

--- a/Packages/com.unity.inputsystem/InputSystem/Editor/ControlPicker/InputControlPickerDropdown.cs
+++ b/Packages/com.unity.inputsystem/InputSystem/Editor/ControlPicker/InputControlPickerDropdown.cs
@@ -65,7 +65,9 @@ namespace UnityEngine.InputSystem.Editor
 
         protected override void OnDestroy()
         {
+            #if UNITY_INPUT_SYSTEM_PROJECT_WIDE_ACTIONS
             InputActionsEditorSettingsProvider.SetIMGUIDropdownVisible(false, false);
+            #endif
             m_RebindingOperation?.Dispose();
             m_RebindingOperation = null;
         }
@@ -121,7 +123,9 @@ namespace UnityEngine.InputSystem.Editor
 
         protected override void ItemSelected(AdvancedDropdownItem item)
         {
+            #if UNITY_INPUT_SYSTEM_PROJECT_WIDE_ACTIONS
             InputActionsEditorSettingsProvider.SetIMGUIDropdownVisible(false, true);
+            #endif
             var path = ((InputControlDropdownItem)item).controlPathWithDevice;
             m_OnPickCallback(path);
         }

--- a/Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/InputActionsEditorSettingsProvider.cs
+++ b/Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/InputActionsEditorSettingsProvider.cs
@@ -1,5 +1,6 @@
 #if UNITY_EDITOR && UNITY_INPUT_SYSTEM_PROJECT_WIDE_ACTIONS
 using System.Collections.Generic;
+using System.Threading.Tasks;
 using UnityEditor;
 using UnityEditor.ShortcutManagement;
 using UnityEngine.UIElements;
@@ -130,6 +131,18 @@ namespace UnityEngine.InputSystem.Editor
 
             m_IMGUIDropdownVisible = visible;
         }
+        
+        private async void DelayFocusLost(bool relatedTargetWasNull)
+        {
+            await Task.Delay(120);
+
+            // We delay this call to ensure that the IMGUI flag has a chance to change first. 
+            if (relatedTargetWasNull && m_HasEditFocus && !m_IMGUIDropdownVisible)
+            {
+                m_HasEditFocus = false;
+                SaveAssetOnFocusLost();
+            }
+        }
 
         private void OnEditFocusLost(FocusOutEvent @event)
         {
@@ -138,11 +151,7 @@ namespace UnityEngine.InputSystem.Editor
             // elements outside of project settings Editor Window. Also note that @event is null when we call this
             // from OnDeactivate().
             var element = (VisualElement)@event?.relatedTarget;
-            if (element == null && m_HasEditFocus && !m_IMGUIDropdownVisible)
-            {
-                m_HasEditFocus = false;
-                SaveAssetOnFocusLost();
-            }
+            DelayFocusLost(element == null);
         }
 
         private void OnStateChanged(InputActionsEditorState newState)

--- a/Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/InputActionsEditorSettingsProvider.cs
+++ b/Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/InputActionsEditorSettingsProvider.cs
@@ -131,12 +131,12 @@ namespace UnityEngine.InputSystem.Editor
 
             m_IMGUIDropdownVisible = visible;
         }
-        
+
         private async void DelayFocusLost(bool relatedTargetWasNull)
         {
             await Task.Delay(120);
 
-            // We delay this call to ensure that the IMGUI flag has a chance to change first. 
+            // We delay this call to ensure that the IMGUI flag has a chance to change first.
             if (relatedTargetWasNull && m_HasEditFocus && !m_IMGUIDropdownVisible)
             {
                 m_HasEditFocus = false;

